compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1637,6 +1637,135 @@ Im2colOp::reifyResultShapes(OpBuilder &b,
+           << getIndexingMapsAttr().size()
+           << ") to be same as the "
+              "number of input/output operands ("
+           << getNumOperands() << ")";
+  }
+
+  // Check the form of the indexing maps.
+  std::optional<unsigned> numSymbolDims;
+  for (auto [index, indexingMapAttr, operand] :
+       llvm::enumerate(getIndexingMapsAttr(), getOperands())) {
+    auto indexingMap = cast<AffineMapAttr>(indexingMapAttr).getValue();
+    if (indexingMap.isEmpty()) {
+      continue;
+    }
+
+    // Domain must be consistent.
+    unsigned numLoops = getNumLoops();
+    if (indexingMap.getNumDims() != numLoops) {
+      return emitOpError("expected indexing_map #")
+             << index << " to have " << numLoops
+             << " dim(s) to match the number of loops or be zero";
+    }
+
+    // Check that number of symbols is consistent.
+    if (numSymbolDims) {
+      if (indexingMap.getNumSymbols() != numSymbolDims.value()) {
+        return emitOpError(
+                   "inconsistent number of symbol dimensions in indexing_map #")
+               << index << ", expected " << numSymbolDims.value()
+               << " instead of " << indexingMap.getNumSymbols();
+      }
+    } else {
+      numSymbolDims = indexingMap.getNumSymbols();
+    }
+
+    // Range must match the rank of the operands.
+    int64_t rank = getRank(operand);
+    if (indexingMap.getNumResults() != rank) {
+      return emitOpError("expected operand rank(")
+             << rank << ") to match the result rank of indexing map #" << index;
+    }
+  }
+
+  // Check that number of basic block arguments is same as number of operands
+  Block *body = getBody();
+  if (body->getNumArguments() != getNumOperands()) {
+    return emitOpError("expected as many basic block arguments (")
+           << body->getNumArguments() << ") as the number of operands ("
+           << getNumOperands() << ")";
+  }
+
+  // Check that type of the basic block argument matches the type of the
+  // operands.
+  for (auto [index, bbArg, operand] :
+       llvm::enumerate(body->getArguments(), getOperands())) {
+    Type operandType = operand.getType();
+    Type bbArgType = bbArg.getType();
+
+    if (operandType.isIntOrIndexOrFloat()) {
+      if (!bbArgType.isIntOrIndexOrFloat()) {
+        return emitOpError("for (scalar) operand #")
+               << index
+               << " expected corresponding basic block argument to be scalar "
+                  "as well";
+      }
+      continue;
+    }
+
+    auto operandTensorType = cast<RankedTensorType>(operandType);
+    auto bbArgTensorType = dyn_cast<RankedTensorType>(bbArgType);
+    if (!bbArgTensorType) {
+      return emitOpError("for (tensor) operand #")
+             << index
+             << " expected corresponding basic block argument to be tensor as "
+                "well";
+    }
+
+    // Check that the basic block arg has same rank/element type, but all shapes
+    // dynamic.
```

Comment:
Shouldn't this be `all non-symbolic shapes dynamic`?

Reply:
Potentially, let me leave it this way for now. Ill relax as needed.